build-system/tasks/dns-monitor.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const argv = require('minimist')(process.argv.slice(2));
+const colors = require('ansi-colors');
+const log = require('fancy-log');
+const readline = require('readline');
+const {execScriptAsync} = require('../common/exec');
+
+const PERMITTED_DOMAIN_REGEXES = [
+  /\.internal$/, // Travis use
+  /^(amp-test-status-bot|amp-error-reporting)\.appspot\.com$/, // Infra team
+  /\.test$/, // .test TLD is safe to use for tests
+];
+
+const TCPDUMP_DNS_REGEX = /^(\d{2}:\d{2}:\d{2})\.\d{6} IP6? [^ ]+ > [^ ]+: \d+\+ (A|AAAA)\? ([^ ]+)\. \(\d+\)$/;
```

Comment:
This is a complicated regex that will be hard to maintain. Can you split it into parts with inline comments for each part? Can you also add a comment above it explaining how it works at a high level?

Reply:
Great idea. Will do.